build_test/lib/src/debug_test_builder.dart

```diff
@@ -5,45 +5,133 @@
+AssetId _debugHtmlId(AssetId dartTest) {
+  return dartTest.changeExtension('.debug.html');
+}
+
+/// Returns the JS script path for the browser for [dartTest].
+String _jsScriptPath(AssetId dartTest) {
+  return '${p.basename(dartTest.path)}.js';
+}
+
+/// Generates a `*.debug.html` for every file in `test/**/*_test.dart`.
+///
+/// This is normally used in order to use Chrome (or another browser's)
+/// debugging tools (such as setting breakpoints) while running a test suite.
+class DebugTestBuilder implements Builder {
+  /// Generates a `.debug.html` for the provided `._test.dart` file.
+  static Future<void> _generateDebugHtml(
+    BuildStep buildStep,
+    AssetId dartTest,
+  ) async {
+    final customHtmlId = _customHtmlId(dartTest);
+    final jsScriptPath = _jsScriptPath(buildStep.inputId);
     String debugHtml;
-    var jsScriptPath = '${p.basename(buildStep.inputId.path)}.js';
     if (await buildStep.canRead(customHtmlId)) {
-      var htmlContent = await buildStep.readAsString(customHtmlId);
-      var document = parse(htmlContent);
+      debugHtml = _replaceCustomHtml(
+        await buildStep.readAsString(customHtmlId),
+        jsScriptPath,
+      );
+    } else {
+      debugHtml = _createDebugHtml(jsScriptPath);
+    }
+    return buildStep.writeAsString(_debugHtmlId(dartTest), debugHtml);
+  }
 
-      // Replace the <link rel="x-dart-test" href="some_path"> tag with
-      var linkTag = document.querySelector('link[rel="x-dart-test"]');
-      var scriptTag = new Element.tag('script');
-      scriptTag.attributes['src'] = jsScriptPath;
-      linkTag.replaceWith(scriptTag);
+  /// Returns the HTML body of [customHtml] modified to work with this package.
+  static String _replaceCustomHtml(String customHtml, String jsScriptPath) {
+    final document = parse(customHtml);
 
-      // Remove the <script src="packages/test/dart.js"></script> if present.
-      document.querySelector('script[src="packages/test/dart.js"]')?.remove();
+    // Replace <link rel="x-dart-test"> with <script src="{jsScriptPath}">.
+    final linkTag = document.querySelector('link[rel="x-dart-test"]');
+    final scriptTag = new Element.tag('script');
+    scriptTag.attributes['src'] = jsScriptPath;
+    linkTag.replaceWith(scriptTag);
 
-      debugHtml = document.outerHtml;
-    } else {
-      debugHtml = '''
-<html>
-  <head>
-    <script src="$jsScriptPath"></script>
-  </head>
-</html>
-      ''';
+    // Remove the <script src="packages/test/dart.js"></script> if present.
+    document.querySelector('script[src="packages/test/dart.js"]')?.remove();
+
+    return document.outerHtml;
+  }
+
+  static String _createDebugHtml(String jsScriptPath) => ''
+      '<html>\n'
+      '  <head>\n'
+      '    <script src="$jsScriptPath"></script>\n'
+      '  </head>\n'
+      '</html>\n';
+
+  const DebugTestBuilder();
+
+  @override
+  Map<String, List<String>> get buildExtensions {
```

Comment:
[nit] I prefer the `final buildExtensions = const {...}` version.

Reply:
Sorry was debugging. Done.